packages/wdio-shared-store-service/src/client.ts

```diff
@@ -29,38 +39,46 @@ export const getValue = async (key: string): Promise<string | number | boolean |
  * @param {*}       value `store[key]` value (plain object)
  */
 export const setValue = async (key: string, value: JsonCompatible | JsonPrimitive) => {
-    /**
-     * if someone calls `setValue` in `onPrepare` we don't have a base url
-     * set as the launcher is called after user hooks. In this case we need
-     * to wait until it is set and flush all messages.
-     */
-    if (baseUrl) {
+    const setPromise = baseUrlPromise.then((baseUrl) => {
         return got.post(`${baseUrl}/set`, { json: { key, value } }).catch(errHandler)
-    }
+    })
+
+    return isBaseUrlReady ? setPromise : Promise.resolve()
+}
 
-    log.info('Shared store server not yet started, collecting value')
-    pendingValues.set(key, value)
+/**
+ *
+ * @param {string}  key
+ * @param {*}       value
+ */
+export const setResourcePool = async (key: string, value: JsonArray) => {
+    const setPromise = baseUrlPromise.then((baseUrl) => {
```

Comment:
If we `await baseUrlPromise` and `setResourcePool()` is inside an `onPrepare` hook, then it would be stuck forever because this service only sets an url after the users hooks are finished, correct? If this is not true then awaiting in all functions here is way simpler